### PR TITLE
Rupoors fix and support for magic meter upgrade and hurricane spin added

### PR DESCRIPTION
**Pull request by CelestialKitsune**
The byte used for the hurricane spin was found in the original randomizer [code](https://github.com/LagoLunatic/wwrando/blob/15457368280934ade7dc41e4509025753bb917fc/asm/patches/custom_funcs.asm#L819-L831)

### CODE REVIEW

File: Dolphin/windWakerInterface.py [hunk 1]
```diff
@@ -159,7 +163,7 @@ def downgrade_wallet():
 
 def give_rupees(amount: int):
     if random_rupoors and bool(random.getrandbits(1)):
-            amount = ((amount ^ 0xFFFFFFFF) + 1 )
+            amount = -amount + 2**32
```
Comment (Crain-32) [inline Dolphin/windWakerInterface.py]: I'm still not entirely sure why this fix was required. I've tested out this function before the last minor release (0.0.2.2 on February 27), and I haven't heard anything about it. Do you have an example?

File: Dolphin/windWakerInterface.py [hunk 2]
```diff
@@ -263,6 +267,18 @@ def take_pearl(item: int):
...
 
+def give_magic_upgade():
+    dme.write_byte(0x803C4C1B, 0x20)
+    dme.write_byte(0x803C4C1C, 0x20)
+
+def take_magic_upgade():
+    dme.write_byte(0x803C4C1B, 0x10)
+
+def give_hurricane_spin():
+    dme.write_byte(0x803C5295, 1)
```
Comment (Crain-32) [inline Dolphin/windWakerInterface.py]: This is looking like it's actually a single-bit toggle. Can you please rewrite it to be more like the `give_heros_charm` and `take_heros_charm` below? It should just be an offset of `0`

**Review by Crain-32 — CHANGES_REQUESTED** [review]
Two comments left on the PR. One is a question for an example of the issue being resolved. The second is a formating request. Once both those are satisfied I see no reason to not merge.

File: Dolphin/windWakerInterface.py (hunk 2 above)
Comment (CelestialKitsune) [reply Dolphin/windWakerInterface.py]: Also requesting a change for the magic meter upgrade or only for hurricane spin?

File: Dolphin/windWakerInterface.py (hunk 1 above)
Comment (CelestialKitsune) [reply Dolphin/windWakerInterface.py]: My changes were actually not needed. The initial method you wrote works fine. 
I was still on the mindset that it wasn't working as I hadn't heard anything from you about that.

File: Dolphin/windWakerInterface.py (hunk 2 above)
Comment (Crain-32) [reply Dolphin/windWakerInterface.py]: Only the Hurricane Spin. Magic Meter Upgrade is fine.

File: Dolphin/windWakerInterface.py (hunk 1 above)
Comment (Crain-32) [reply Dolphin/windWakerInterface.py]: No worries. If you could remove the changes then that would be great.
The current solution makes the 2's Complement of the Number, so it's mirroring the Hardware's Expectations.
Your solution likely does the same, it's just not as clear.

Comment (CelestialKitsune) [reply Dolphin/windWakerInterface.py]: Fixed in [957f6c7](https://github.com/Crain-32/Multiworld_Client/pull/1/commits/957f6c7947fb7ceab256b9e828312e72b6775918)

File: Dolphin/windWakerInterface.py (hunk 2 above)
Comment (CelestialKitsune) [reply Dolphin/windWakerInterface.py]: Fixed in [c0e5e7f](https://github.com/Crain-32/Multiworld_Client/pull/1/commits/c0e5e7f96793267eabcc02a49496e5dc2ecacd27)

**Review by Crain-32 — APPROVED** [review]

**Pull request merged**